src/community/rest-upload/src/main/java/org/geoserver/restupload/ResumableUploadPathMapper.java

```diff
@@ -39,10 +39,7 @@ public void mapItemPath(String workspace, String store, Map<String, String> stor
         }
         String root = RESTUtils.getRootDirectory(workspace, store, catalog);
         if (root == null || root.isEmpty()) {
-            //GEOSERVER_DATA_DIR/data/
-            GeoServerResourceLoader loader = GeoServerExtensions.bean(GeoServerResourceLoader.class);
-            Resource data = loader.get("data");
-            root = data.dir().getAbsolutePath();
+            root = "data";
```

Comment:
@NielsCharlier I know you did not make the original poor decision, but this should be refactored as a CONSTANT_WITH_A_MEANINGFUL_NAME that can be shared and commented. Bare strings as path names are a no-no.